### PR TITLE
Install Yarn, React & Webpack

### PR DESCRIPTION
**Pull request by whyderrick**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -17,7 +16,7 @@
     end
 
   get '/sign_in' => 'clearance/sessions#new', as: 'sign_in'
-  delete '/sign_out' => 'clearance/session#delete', as: 'sign_out'
+  delete '/sign_out' => 'clearance/sessions#destroy', as: 'sign_out'
```
[the remark this answers is not in the document]
Comment (whyderrick) [reply config/routes.rb]: handled in another branch 😝

**Review by gylaz — APPROVED** [review]
Looks good, but I've never setup Webpack on a Rails project.

Might be nice to branch this off the other PR that renames the app (or merge that in first before opening this one), as to not dirty the diff with the rename changes.

File: webpack/config.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+console.log("current directory is " + __dirname);
```
Comment (thorncp) [inline webpack/config.js]: Does this log statement need to be here?

File: webpack/config.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+const path = require("path");
+
+module.exports = {
+  entry: "./webpack/app/foo.js",
```
Comment (thorncp) [inline webpack/config.js]: It seems this file doesn't exist. Should this point to a real file? Or, can this setting be removed?

Comment (jcmorrow) [inline webpack/config.js]: `foo`?

File: webpack/config.js (hunk 2 above)
Comment (jcmorrow) [inline webpack/config.js]: Was this just here for debugging?

**Review by jcmorrow — APPROVED** [review]
Couple of questions, overall looks good.

**Review by thorncp — APPROVED** [review]
:cat:

File: webpack/config.js (hunk 2 above)
Comment (whyderrick) [reply webpack/config.js]: Yup. Thanks for catching it.

Comment (whyderrick) [reply webpack/config.js]: not at all. thanks.

File: webpack/config.js (hunk 3 above)
Comment (whyderrick) [reply webpack/config.js]: leftover from testing this. Renamed to `main.js` now. Thanks.

**Pull request merged**